pkg/backend/snapshot.go

```diff
@@ -295,6 +348,15 @@ func (sm *SnapshotManager) markNew(state *resource.State) {
 	logging.V(9).Infof("Appended new state snapshot to be written: %v", state.URN)
 }
 
+// markAbandoned marks a resource as abandoned. A resource that has been abandoned will not be
```

Comment:
I think that this comment could be made clearer with two minor adjustments:
- `s/resource/state`
- provider an example of the `Update` case.

As written, it's easy to interpret the fact that we abandon the new state of a failed update step will omit the resource from the final snapshot.